### DISCUSSION TITLE
How to share photo with other app on Android?

### DISCUSSION THREAD

**Discussion by raxod502** [photoprism/photoprism#3907] [Q&A] [answered]
Hi, I'm working on finalizing my shutdown of Google Photos and one of the last issues I'm running into is how to share a photo from my library with someone in another app (e.g. Signal, Fastmail). Since Google Photos has a native app on Android, it is able to activate the "share" context menu which is the general standard on Android for transferring data between different apps. However, with PhotoPrism, since it lives in the browser, it seems like I don't have a way of activating that menu (at least in Firefox). Instead, I have to click the download button, then use a different app like Photo Editor to open the file from my downloads folder and activate the share menu.

This is all doable, but I feel very silly doing it. Is there a better way that I didn't think of? Or what do other people do when they want to send photos to other people? (I know copying the URL is another option but in many contexts it makes more sense to send an image directly rather than a link to one.)

**Comment by raxod502** [comment]
Okay, well, I am shocked and a little betrayed by Android, apparently the only reason this did not work in the first place was because the option is randomly disabled when you use the "install website on home screen" functionality! See screenshots below for a comparison of the exact same page in the normal Firefox browser versus in what is apparently the restricted mode.

Now I'd rather *like* to have a dedicated icon on my home screen for PhotoPrism so if anyone knows a workaround to prevent Firefox from not letting you share images, I'm all ears...

<img src="https://github.com/photoprism/photoprism/assets/6559064/57e3691f-6ab0-416d-bbe7-b785d97910bd" width="200"/>
<img src="https://github.com/photoprism/photoprism/assets/6559064/5aea5195-bdfe-4a1e-b4d2-9d8d5e83fe01" width="200"/>

**Reply by raxod502** [reply]
Seems like a Firefox issue probably, so I asked at https://support.mozilla.org/en-US/questions/1431859

**Reply by graciousgrey** [reply, marked as answer]
You might also be interested in this Android client :)

https://play.google.com/store/apps/details?id=ua.com.radiokot.photoprism
https://github.com/Radiokot/photoprism-android-client

**Reply by raxod502** [reply]
Welp, that wholly solves my use case. I wish I knew about it earlier!
